lgn-messages/src/lib.rs

```diff
@@ -1,7 +1,450 @@
+                ProverType::V1Query => "V1Query",
+                ProverType::V1Groth16 => "V1Groth16",
+            }
+        )
+    }
+}
+
+pub trait ToProverType {
+    fn to_prover_type(&self) -> ProverType;
+}
+
+const UNSUPPORTED: &str = "unsupported";
+const GROTH16: &str = "groth16";
+const BATCHED_INDEX_EXTRACTION: &str = "batched_index_extraction";
+const BATCHED_LENGTH_EXTRACTION: &str = "batched_length_extraction";
+const BATCHED_CONTRACT_EXTRACTION: &str = "batched_contract_extraction";
+const BLOCK_EXTRACTION: &str = "block_extraction";
+const CONTRACT_LEAF_EXTRACTION: &str = "contract_leaf_extraction";
+const CONTRACT_EXTENSION_EXTRACTION: &str = "contract_extension_extraction";
+const CONTRACT_BRANCH_EXTRACTION: &str = "contract_branch_extraction";
+const LENGTH_BRANCH_EXTRACTION: &str = "length_branch_extraction";
+const LENGTH_EXTENSION_EXTRACTION: &str = "length_extension_extraction";
+const LENGTH_LEAF_EXTRACTION: &str = "length_leaf_extraction";
+const SINGLE_VALUE_LEAF_EXTRACTION: &str = "single_value_leaf_extraction";
+const MAPPING_VALUE_LEAF_EXTRACTION: &str = "mapping_value_leaf_extraction";
+const MAPPING_OF_MAPPING_LEAF_EXTRACTION: &str = "mapping_of_mapping_leaf_extraction";
+const VALUE_EXTENSION_EXTRACTION: &str = "value_extension_extraction";
+const VALUE_BRANCH_EXTRACTION: &str = "value_branch_extraction";
+const FINAL_SIMPLE_EXTRACTION: &str = "final_simple_extraction";
+const FINAL_LENGTHED_EXTRACTION: &str = "final_lengthed_extraction";
+const FINAL_MERGE_EXTRACTION: &str = "final_merge_extraction";
+const FINAL_DUMMY_EXTRACTION: &str = "final_dummy_extraction";
+const CELLS_TREE_LEAF_EXTRACTION: &str = "cells_tree_leaf_extraction";
+const CELLS_TREE_FULL_NODE_EXTRACTION: &str = "cells_tree_full_node_extraction";
+const CELLS_TREE_PARTIAL_NODE_EXTRACTION: &str = "cells_tree_partial_node_extraction";
+const ROW_TREE_LEAF_EXTRACTION: &str = "row_tree_leaf_extraction";
+const ROW_TREE_FULL_EXTRACTION: &str = "row_tree_full_extraction";
+const ROW_TREE_PARTIAL_EXTRACTION: &str = "row_tree_partial_extraction";
+const BLOCK_TREE_LEAF_EXTRACTION: &str = "block_tree_leaf_extraction";
+const BLOCK_TREE_PARENT_EXTRACTION: &str = "block_tree_parent_extraction";
+const BLOCK_TREE_MEMBERSHIP_EXTRACTION: &str = "block_tree_membership_extraction";
+const IVC_FIRST_PROOF: &str = "ivc_first_proof";
+const IVC_SUBSEQUENT_PROOF: &str = "ivc_subsequent_proof";
+const QUERY_BATCHED_TABULAR: &str = "query_batched_tabular";
+const QUERY_ROW_CHUNK_WITH_AGGREGATION: &str = "query_row_chunk_with_aggregation";
+const QUERY_CHUNK_AGGREGATION: &str = "query_chunk_aggregation";
+const QUERY_NON_EXISTENCE: &str = "query_non_existence";
+const UNIVERSAL_QUERY_WITH_AGGREGATION: &str = "universal_query_with_aggregation";
+const UNIVERSAL_QUERY_NO_AGGREGATION: &str = "universal_query_no_aggregation";
+const REVELATION_NO_RESULTS_TREE: &str = "revelation_no_results_tree";
+const REVELATION_UNPROVEN_OFFSET: &str = "revelation_unproven_offset";
```

Comment:
Added metrics for all message types. This should is also useful for the QE and should remove duplicated code.